robots/commenter/main.go

```diff
@@ -155,6 +157,7 @@ func makeQuery(query string, includeArchived, includeClosed, includeLocked bool,
 type client interface {
 	CreateComment(owner, repo string, number int, comment string) error
 	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+	ListIssueComments(org, repo string, number int) ([]github.IssueComment, error)
```

Comment:
I'm worried about the token consumption. This is a paginated api with a max of 100 results, so on longer issues it may be multiple API calls per PR.

Reply:
Are there any alternatives for this API? Is this something that in your opinion is a blocker for this feature?

Reply:
It's not a blocker for having the feature IMO, but potential blocker to using the feature (I'm not sure if we could afford to use it in OpenShift, for example).

I don't think there's an alternative API for this specific check. Most of Prow avoids doing this kind of checks because they are API-expensive, and Search API is the preferred way how to obtain sets of PRs/issues that conform to some criteria. Of course the Search API capabilities are limited.

I think this PR is simply a bandaid for `retester` being a dump Search->Comment wrapper over GH API. What we *really* need is 1) to make retester only comment on PRs where it makes sense (I think we would need to integrate it with Tide config to work over actual merge criteria queries, so it could do its searches based on that) and 2) find some API-friendly way to implement some retester backoff (which is IMO more desirable than a simple ceiling).

Reply:
I agree. I took the current implementation and patched it in order to cover those gaps, but probably the `retester` job/bot should just be more sophisticated. I'll look at how it could be done.

Reply:
I never looked into it too close, but I always thought retesting could be an (optional?) feature of Tide itself. It knows the merge criteria, so it could search for all PRs that would be mergeable if not for failing tests (so retest users would not need to do the guesswork [like this](https://github.com/openshift/release/blob/b36a66636ee955d500edd29ad8c86e670e149067/ci-operator/jobs/infra-periodics.yaml#L212-L216) to specify the retest criteria). It could also optionally store some limited state (IIRC it already does - for storing history) for the retest backoff.

Reply:
Could you open a ticket for that?